soh/src/overlays/actors/ovl_En_Ge1/z_en_ge1.c

```diff
@@ -533,20 +534,28 @@ void EnGe1_WaitTillItemGiven_Archery(EnGe1* this, GlobalContext* globalCtx) {
                         break;
                 }
             } else {
-                getItemId = Randomizer_GetItemFromKnownCheck(RC_GF_HBA_1500_POINTS, CUR_UPG_VALUE(UPG_QUIVER) == 1 ? GI_QUIVER_40 : GI_QUIVER_50).getItemId;
+                getItemEntry = Randomizer_GetItemFromKnownCheck(RC_GF_HBA_1500_POINTS, CUR_UPG_VALUE(UPG_QUIVER) == 1 ? GI_QUIVER_40 : GI_QUIVER_50);
+                getItemId = getItemEntry.getItemId;
             }
         } else {
             if (!gSaveContext.n64ddFlag) {
                 getItemId = GI_HEART_PIECE;
             } else {
-                getItemId = Randomizer_GetItemFromKnownCheck(RC_GF_HBA_1000_POINTS, GI_HEART_PIECE).getItemId;
+                getItemEntry = Randomizer_GetItemFromKnownCheck(RC_GF_HBA_1000_POINTS, GI_HEART_PIECE);
+                getItemId = getItemEntry.getItemId;
             }
         }
-        func_8002F434(&this->actor, globalCtx, getItemId, 10000.0f, 50.0f);
+
+        if (!gSaveContext.n64ddFlag) {
```

Comment:
Logical Error: Can it get to this point without setting getItemEntry?

Reply:
Same fix as I suggested for fishing should work here.